components/org.wso2.carbon.database.utils/src/main/java/org/wso2/carbon/database/utils/jdbc/JdbcTemplate.java

```diff
@@ -438,6 +461,15 @@ private int doInternalUpdate(QueryFilter queryFilter, PreparedStatement prepared
         return preparedStatement.executeUpdate();
     }
 
+    private int[] doInternalBatchUpdateAndReturnResults(QueryFilter queryFilter, PreparedStatement
```

Comment:
Shall we remove the `...AndReturnResults` part from the function name?

Reply:
We already have a private method called `doInternalBatchUpdate` which does not return the results. Added this to the function name to differentiate it.

Reply:
Ok. Since the logic is same shall we check whether we can update the return type to the existing method?

Reply:
The doInternalBatchUpdate method is utilized by `executeBatchInsert` and `executeBatchDelete` which are public methods. 

Additionally we have already two methods `executeUpdate` and `executeUpdateWithAffectedRows`
Therefore, the convention followed in this class seems to be to have separate methods when a new method is required to return results.

Therefore, IMO it is better to have two separate methods and not modify the existing method doInternalBatchUpdate method.